test/clj_kondo/analysis_test.clj

```diff
@@ -114,6 +114,15 @@
       (assert-submaps
        '[{:name "kw" :reg user/mydef}]
        (:keywords a))))
+  (testing "var usage in re-frame.core/reg-event-db body makes reg available in :context under :in-reg key"
+    (let [a (analyze "(require '[re-frame.core :as rf])
+                      (rf/reg-event-db ::a (constantly {}))"
+                     {:config {:output {:analysis {:keywords true}}}})]
+      (assert-submaps
+       '[{:name require}
+         {:name constantly :context {:in-reg {:k :a :reg re-frame.core/reg-event-db}}}
```

Comment:
- `:k`: can we use a more descriptive name here? e.g. `:keyword`?
- don't we need to save more information than `:a`? E.g. we cannot distinguish between auto-resolved keywords or simple keywords here?
- why don't we just use as the key into the context?

```
{:context {re-frame.core/reg-event-db {:keyword {:name "a" :ns ... :auto-resolved true}
```

Reply:
- re `:k` sure, sounds better
- sure, maybe ns as well
- hm.. not sure it would be easy to work with a context where the key is `re-frame.core/reg-event-db`. In the client code I would rather work with some constant key name I can easily check if there or not. maybe I don't fully understand your intentions here?!

Reply:
@benedekfazekas Another idea:

- We could maybe refer to the keyword using an identifier. This identifier points to the keyword in the `:keywords` analysis. This way we avoid problems of pre-analyzing the keyword for aliases etc, to get the namespace which is done in a later stage.

- Adding to the context should preferably happen using a qualified keyword, so each kind of hook doesn't overwrite keys of other than its own.  I'm find with adding a generic key, but let's use something with `re-frame` in the name. E.g. `{:context {:re-frame/reg {:keyword {:id "123"}}}}`

Reply:
i deffo like the qualified keyword idea

Reply:
@benedekfazekas OK, so what about the value for the key? Just adding `:keyword :a` there isn't sufficient, I think.

Reply:
worked on the shape of the data, hopefully along he lines of this discussion ;) let me know what you think

Reply:
+1 about adding an id to the keyword, it's similar to how `:locals` works today, the issue is that we'd need to add ids for all keywords but won't have issues about something that was not analyzed yet